taichi/backends/device.h

```diff
@@ -303,6 +310,20 @@ class CommandList {
                                const BufferImageCopyParams &params) {
     TI_NOT_IMPLEMENTED
   }
+  virtual void copy_image(DeviceAllocation dst_img,
+                          DeviceAllocation src_img,
+                          ImageLayout dst_img_layout,
+                          ImageLayout src_img_layout,
+                          const ImageCopyParams &params) {
+    TI_NOT_IMPLEMENTED
```

Comment:
nit: If this is a pure virtual class, make all these methods `=0`?

Reply:
That might make implementation of each device a little bit more complicated if they don't support something